code generators/de.fhdo.lemma.model_processing.code_generation.container_base/src/main/java/de/fhdo/lemma/model_processing/code_generation/container_base/code_generation/BuildScriptCodeGenerator.xtend

```diff
@@ -0,0 +1,99 @@
+package de.fhdo.lemma.model_processing.code_generation.container_base.code_generation
+
+import de.fhdo.lemma.model_processing.annotations.CodeGenerationModule
+import de.fhdo.lemma.model_processing.builtin_phases.code_generation.AbstractCodeGenerationModule
+import de.fhdo.lemma.operation.intermediate.IntermediatePackage
+import de.fhdo.lemma.model_processing.languages.LanguageDescription
+import de.fhdo.lemma.operation.intermediate.IntermediateOperationModel
+import de.fhdo.lemma.model_processing.code_generation.container_base.util.Util
+import de.fhdo.lemma.model_processing.code_generation.container_base.buildscript.OpendRootBuildScript
+import de.fhdo.lemma.model_processing.code_generation.container_base.template.BuildScriptTemplate
+import java.util.HashMap
+import java.util.Map
+import java.io.File
+import org.jetbrains.annotations.NotNull
+
+/**
+ * Main class of the build script code generation module of the container base code generator.
+ *
+ * @author [Philip Wizenty](mailto:philip.wizenty@fh-dortmund.de)
+ */
+@CodeGenerationModule(name="buildscript")
+class BuildScriptCodeGenerator extends AbstractCodeGenerationModule {
+    var IntermediateOperationModel model
+    var Map<String, String> content
+
+    /*
+     * This method is responsible for handling the instance of the intermediate operation model.
+     * The corresponding functions are called depending on whether the operation node is an instance
+     * of an intermediate container or infrastructure node.
+     */
+    override execute(String[] arg0, String[] arg1) {
+        // Receive the intermediate operation model
+        model = this.intermediateModelResource.contents.get(0) as IntermediateOperationModel
+
+        // Create a Map which contains the generated source code
+        content = new HashMap
+
+        OpendRootBuildScript.instance.loadExistingRootBuildScript(targetFolder)
+
+        // Create a build script for containers
+        model.containers.forEach[container |
+            val nodeName = Util::buildPathFromQualifiedName(
+                container.deployedServices?.get(0).qualifiedName
+            )
+            this.createNodeSpecificBuildScripts(nodeName.toLowerCase, container.name.toLowerCase)
+            this.createRootBuildScript(nodeName.toLowerCase)
+        ]
+
+        // Create build script for infrastructure nodes
+        model.infrastructureNodes.forEach[node |
+            val property = node.defaultValues.findFirst[property |
+                property.technologySpecificProperty.name == "imageType"]
+            val nodeName = node.name.toLowerCase
+
+            if (property.value.toLowerCase != "dockerhub") {
```

Comment:
Please add a clarifying value. Why is the value `dockerhub` "magic" in this instance?